### ISSUE TITLE
Improvement computation of the log_likelihood

### ISSUE THREAD

**Issue by jbmouret** [resibots/limbo#9]
In the log_likelihood of gp_auto, we could avoid to recompute alpha and use this->_alpha instead.
```
 -        // alpha = K^{-1} * this->_obs_mean;
 -        Eigen::VectorXd alpha = this->_llt.matrixL().solve(this->_obs_mean);
 -        this->_llt.matrixL().adjoint().solveInPlace(alpha);
 
         // alpha * alpha.transpose() - K^{-1}
 -        w = alpha * alpha.transpose() - w;
 +        w = this->_alpha * this->_alpha.transpose() - w;
```
What do you think?

**Comment by fedeallocati** [comment]
Makes completly sense

**Comment by Aneoshun** [comment]
Just be careful about the likelihood optimization, which compute the likelihood in parallel.
So be sure that the optimization uses copies of the GP and does not simply call the method.

Moreover, the alpha value changes according to the hyper parameters, thus be sure that alpha is correctly updated during the optimization process. 




Antoine CULLY
Phd-Candidate in Robotics and Artificial Intelligence
DGA-UPMC
Institut des Systèmes Intelligents et de Robotique
http://pages.isir.upmc.fr/~cully/website/

**Comment by fedeallocati** [comment]
I think in this case it's okay, because the likelihood computation also use another member variables that change with the hyperparameters, like the LLT decomposition of the kernel matrix, and it's working okay in parallel

**Comment by costashatz** [comment]
Yeap. It should be OK. Because in `rprop`, the `log_likelihood` and `log_likelihood_grad` functions are called serially. The former sets `this->_alpha` and so I believe the latter can use it. And also ( @Aneoshun ) the GP is always copied and only the copy is used in `rprop`.

I will do it after we have merged my current PR.

**Comment by costashatz** [comment]
Fixed in 051281d04934f1d7a37fcedb978aa61e4c3773c7!
